snorkel/contrib/lstm/lstm_base.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+import random
+import tensorflow as tf
+import tensorflow.contrib.rnn as rnn
+
+from snorkel.learning import LabelBalancer, TFNoiseAwareModel
+from utils import f1_score, get_bi_rnn_output, SymbolTable
+from time import time
+
+
+SD = 0.1
+
+
+class LSTMBase(TFNoiseAwareModel):
+
+    representation = True
+
+    def __init__(self, save_file=None, name='LSTMBase', seed=None, n_threads=4):
+        """Base class for LSTM"""
+        # Define metadata
+        self.mx_len    = None # Max sentence length
+        self.dim       = None # Embedding dimension
+        self.n_v       = None # Vocabulary size
+        self.lr        = None # Learning rate
+        self.attn      = None # Attention window
+        self.word_dict = SymbolTable() # Symbol table for dictionary
+        # Define input layers
+        self.sentences        = None
+        self.sentence_lengths = None
+        self.train_marginals  = None
+        self.keep_prob        = None
+        self.seed             = seed
+        # Super constructor
+        super(LSTMBase, self).__init__(
+            n_threads=n_threads, save_file=save_file, name=name
+        )
+
+    def _preprocess_data(self, candidates, extend):
+        raise NotImplementedError()
+
+    def _make_tensor(self, x):
+        """Construct input tensor with padding"""
+        batch_size = len(x)
+        x_batch    = np.zeros((batch_size, self.mx_len), dtype=np.int32)
+        len_batch  = np.zeros(batch_size, dtype=np.int32)
+        for j, token_ids in enumerate(x):
+            t               = min(len(token_ids), self.mx_len)
```

Comment:
Yeah punted on this, but will add an error in the preprocessing steps of the subclasses